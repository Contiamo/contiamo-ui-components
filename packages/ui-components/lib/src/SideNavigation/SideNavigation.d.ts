/// <reference types="react" />
import * as React from "react";
import SideNavigationHeader from "./Header/SideNavigationHeader";
import SideNavigationItem from "./Item/SideNavigationItem";
import SideNavigationLink from "./Link/SideNavigationLink";
export interface IProps {
    key?: string | number;
    css?: {};
    className?: string;
    children?: React.ReactNode;
    color?: string;
    expandOnHover?: boolean;
    expandedWidth?: number;
    width?: number;
    fix?: boolean;
}
declare const SideNavigation: React.SFC<IProps>;
export default SideNavigation;
export { SideNavigationHeader, SideNavigationItem, SideNavigationLink };
