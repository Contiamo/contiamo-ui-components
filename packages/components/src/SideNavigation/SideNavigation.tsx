import * as React from "react"
import glamorous, { GlamorousComponent } from "glamorous"
import { hexOrColor, readableTextColor } from "@operational/utils"
import { Theme } from "@operational/theme"

import SideNavigationHeader from "./Header/SideNavigationHeader"
import SideNavigationItem from "./Item/SideNavigationItem"
import SideNavigationLink from "./Link/SideNavigationLink"

export interface IProps {
  id?: string | number
  css?: {}
  className?: string
  children?: React.ReactNode
  color?: string
  expandOnHover?: boolean
  expandedWidth?: number
  width?: number
  fix?: boolean
}

const Container = glamorous.div(
  ({
    theme,
    color,
    fix,
    expandOnHover,
    expandedWidth,
    width
  }: {
    theme: Theme
    color?: string
    fix?: boolean
    expandOnHover?: boolean
    expandedWidth: number
    width: number
  }): any => {
    const backgroundColor = color
      ? hexOrColor(color)(theme.colors.palette[color])
      : theme.colors.usage.sideNavigationBackground
    const hoverWidth = expandOnHover
      ? {
          transition: ".3s width cubic-bezier(.8, 0, 0, 1)",
          willChange: "width",
          "&:hover": {
            width: expandedWidth
          }
        }
      : {}

    return {
      width,
      backgroundColor,
      position: fix ? "fixed" : "relative",
      zIndex: theme.baseZIndex + 100,
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start",
      height: "100vh",
      overflow: "hidden",
      boxShadow: "1px 0 2px rgba(0, 0, 0, 0.2)",
      color: readableTextColor(backgroundColor)(["black", "white"]),
      ...hoverWidth,
      "& a:focus": {
        outline: 0,
        backgroundColor: "rgba(255, 255, 255, 0.07)"
      }
    }
  }
)

const SideNavigation: React.SFC<IProps> = ({
  id,
  css,
  className,
  children,
  color,
  fix,
  expandOnHover,
  expandedWidth,
  width
}: IProps) => (
  <Container
    key={id}
    css={css}
    className={className}
    color={color}
    fix={fix}
    expandOnHover={expandOnHover}
    expandedWidth={expandedWidth || 240}
    width={width || 60}
  >
    {children}
  </Container>
)

export default SideNavigation
export { SideNavigationHeader, SideNavigationItem, SideNavigationLink }
