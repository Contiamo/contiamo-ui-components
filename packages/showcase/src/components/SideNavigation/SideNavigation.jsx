import React from "react"
import glamorous from "glamorous"
import { Link } from "react-router-dom"
import { Div, Img } from "glamorous"
import { Box, BarChart2, Grid } from "react-feather"

import {
  SideNavigation,
  SideNavigationHeader,
  SideNavigationItem,
  SideNavigationLink
} from "contiamo-ui-components"

const style: {} = ({ theme }) => ({
  "& a": {
    display: "flex",
    alignItems: "center",
    textDecoration: "none",
    color: "white"
  },
  "& .SideNavigationItem__label": {
    marginLeft: 25
  }
})
//expandedShadow => 3px 0 11px rgba(0, 0, 0, 0.19)
// tooltip => 222
// card headers
//lineheight 20, color #747474, fontsize: 13, proximanova400
// card tabs (tabs)
// buttons, 2px radius, bordercolor #e2e2e2, no shadow, inherit color <- copy shadows
//cliuckable things have  a 2px borderraduus

const AppSideNavigation = ({ location }: { location: {} }) =>
  <SideNavigation css={style} fix expandOnHover color="#4E5665">
    <SideNavigationHeader tooltip="Contiamo" size={24}>
      <Img
        css={{ maxWidth: 32, marginRight: 16 }}
        alt="Contiamo"
        src="/img/logo/outline.png"
      />
      Contiamo
    </SideNavigationHeader>

    <SideNavigationItem
      active={location.pathname.match(/\/components/)}
      tooltip="Components"
    >
      <Link to="/components">
        <Box
          color={location.pathname.match(/\/components/) ? "#67FFAA" : "#fff"}
          size={20}
        />
        <Div
          css={{
            color: location.pathname.match(/\/components/) ? "#67FFAA" : "#fff"
          }}
          className="SideNavigationItem__label"
        >
          Components
        </Div>
      </Link>
    </SideNavigationItem>
    <SideNavigationItem tooltip="Composed">
      <Link to="#">
        <Grid color="#fff" size={20} />
        <div className="SideNavigationItem__label">Composed</div>
      </Link>
    </SideNavigationItem>
    <SideNavigationItem tooltip="Visualizations">
      <Link to="#">
        <BarChart2 color="#fff" size={20} />
        <div className="SideNavigationItem__label">Visualizations</div>
      </Link>
    </SideNavigationItem>
    <Div css={{ flexGrow: 1, height: "100%" }} />
  </SideNavigation>

export default AppSideNavigation
