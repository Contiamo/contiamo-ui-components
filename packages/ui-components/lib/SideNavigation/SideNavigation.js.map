{"version":3,"file":"SideNavigation.js","sourceRoot":"","sources":["../../src/SideNavigation/SideNavigation.tsx"],"names":[],"mappings":";;;;;;;;;;AAAA,6BAA8B;AAC9B,uCAAyD;AAGzD,sEAAgE;AA8FvD,+BA9FF,8BAAoB,CA8FE;AA7F7B,gEAA0D;AA6F3B,6BA7FxB,4BAAkB,CA6FwB;AA5FjD,gEAA0D;AA4FP,6BA5F5C,4BAAkB,CA4F4C;AA1FrE,uDAAiE;AAcjE,IAAM,SAAS,GAAG,mBAAS,CAAC,GAAG,CAC7B,UAAC,EAcA;QAbC,gBAAK,EACL,gBAAK,EACL,YAAG,EACH,gCAAa,EACb,gCAAa,EACb,gBAAK;IASL,IAAM,eAAe,GAAG,KAAK;QAC3B,CAAC,CAAC,8BAAU,CAAC,KAAK,CAAC,CAAC,KAAK,CAAC,MAAM,CAAC,OAAO,CAAC,KAAK,CAAC,CAAC;QAChD,CAAC,CAAC,KAAK,CAAC,MAAM,CAAC,KAAK,CAAC,wBAAwB,CAAA;IAC/C,IAAM,UAAU,GAAG,aAAa;QAC9B,CAAC,CAAC;YACE,UAAU,EAAE,qCAAqC;YACjD,UAAU,EAAE,OAAO;YACnB,SAAS,EAAE;gBACT,KAAK,EAAE,aAAa;aACrB;SACF;QACH,CAAC,CAAC,EAAE,CAAA;IAEN,MAAM,YACJ,KAAK,OAAA;QACL,eAAe,iBAAA,EACf,QAAQ,EAAE,GAAG,CAAC,CAAC,CAAC,OAAO,CAAC,CAAC,CAAC,UAAU,EACpC,MAAM,EAAE,KAAK,CAAC,UAAU,GAAG,GAAG,EAC9B,OAAO,EAAE,MAAM,EACf,aAAa,EAAE,QAAQ,EACvB,UAAU,EAAE,YAAY,EACxB,MAAM,EAAE,OAAO,EACf,QAAQ,EAAE,QAAQ,EAClB,SAAS,EAAE,8BAA8B,EACzC,KAAK,EAAE,qCAAiB,CAAC,eAAe,CAAC,CAAC,CAAC,OAAO,EAAE,OAAO,CAAC,CAAC,IAC1D,UAAU,IACb,WAAW,EAAE;YACX,OAAO,EAAE,CAAC;YACV,eAAe,EAAE,2BAA2B;SAC7C,IACF;AACH,CAAC,CACF,CAAA;AAED,IAAM,cAAc,GAAsB,UAAC,EAUlC;QATP,UAAE,EACF,YAAG,EACH,wBAAS,EACT,sBAAQ,EACR,gBAAK,EACL,YAAG,EACH,gCAAa,EACb,gCAAa,EACb,gBAAK;IACO,OAAA,CACZ,oBAAC,SAAS,IACR,GAAG,EAAE,EAAE,EACP,GAAG,EAAE,GAAG,EACR,SAAS,EAAE,SAAS,EACpB,KAAK,EAAE,KAAK,EACZ,GAAG,EAAE,GAAG,EACR,aAAa,EAAE,aAAa,EAC5B,aAAa,EAAE,aAAa,IAAI,GAAG,EACnC,KAAK,EAAE,KAAK,IAAI,EAAE,IAEjB,QAAQ,CACC,CACb;AAba,CAab,CAAA;AAED,kBAAe,cAAc,CAAA"}