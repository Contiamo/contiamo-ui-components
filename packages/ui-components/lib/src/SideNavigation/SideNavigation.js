"use strict";
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorous_1 = require("glamorous");
var SideNavigationHeader_1 = require("./Header/SideNavigationHeader");
exports.SideNavigationHeader = SideNavigationHeader_1.default;
var SideNavigationItem_1 = require("./Item/SideNavigationItem");
exports.SideNavigationItem = SideNavigationItem_1.default;
var SideNavigationLink_1 = require("./Link/SideNavigationLink");
exports.SideNavigationLink = SideNavigationLink_1.default;
var contiamo_ui_utils_1 = require("contiamo-ui-utils");
var style = function (_a) {
    var theme = _a.theme, color = _a.color, fix = _a.fix, expandOnHover = _a.expandOnHover, 
    // for some reason, glamorous doesn't get the defaultProps...
    _b = _a.expandedWidth, 
    // for some reason, glamorous doesn't get the defaultProps...
    expandedWidth = _b === void 0 ? 240 : _b, _c = _a.width, width = _c === void 0 ? 60 : _c;
    var backgroundColor = color ? contiamo_ui_utils_1.hexOrColor(color)(theme.colors.palette[color]) : theme.colors.palette.grey80, hoverWidth = expandOnHover
        ? {
            transition: ".3s width cubic-bezier(.8, 0, 0, 1)",
            willChange: "width",
            "&:hover": {
                width: expandedWidth
            },
            "& .Tooltip": {
                display: "none"
            },
            "&:not(:hover) .SideNavigationHeader::after": {
                content: "none"
            },
            "&:not(:hover) .SideNavigationHeader__options": {
                display: "none"
            }
        }
        : {};
    return __assign({ width: width,
        backgroundColor: backgroundColor, position: fix ? "fixed" : "relative", zIndex: theme.baseZIndex + 100, display: "flex", flexDirection: "column", alignItems: "flex-start", height: "100vh", overflow: "hidden", boxShadow: "1px 0 2px rgba(0, 0, 0, 0.2)", color: contiamo_ui_utils_1.readableTextColor(backgroundColor)(["black", "white"]) }, hoverWidth);
};
var SideNavigation = function (_a) {
    var className = _a.className, children = _a.children;
    return React.createElement("div", { className: className }, children);
};
SideNavigation.defaultProps = {
    expandOnHover: false,
    expandedWidth: 280,
    width: 64,
    fix: false
};
exports.default = glamorous_1.default(SideNavigation)(style);
//# sourceMappingURL=SideNavigation.js.map