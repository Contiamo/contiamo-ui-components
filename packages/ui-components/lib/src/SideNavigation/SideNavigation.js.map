{"version":3,"file":"SideNavigation.js","sourceRoot":"","sources":["../../../src/SideNavigation/SideNavigation.tsx"],"names":[],"mappings":";;;;;;;;;;AAAA,6BAA8B;AAE9B,uCAAiC;AAEjC,sEAAgE;AA8EvD,+BA9EF,8BAAoB,CA8EE;AA7E7B,gEAA0D;AA6E3B,6BA7ExB,4BAAkB,CA6EwB;AA5EjD,gEAA0D;AA4EP,6BA5E5C,4BAAkB,CA4E4C;AA1ErE,uDAAiE;AAcjE,IAAM,KAAK,GAAG,UAAC,EASP;QARN,gBAAK,EACL,gBAAK,EACL,YAAG,EACH,gCAAa;IAEb,6DAA6D;IAC7D,qBAAmB;IADnB,6DAA6D;IAC7D,wCAAmB,EACnB,aAAU,EAAV,+BAAU;IAEV,IAAM,eAAe,GAAG,KAAK,GAAG,8BAAU,CAAC,KAAK,CAAC,CAAC,KAAK,CAAC,MAAM,CAAC,OAAO,CAAC,KAAK,CAAC,CAAC,GAAG,KAAK,CAAC,MAAM,CAAC,OAAO,CAAC,MAAM,EAC1G,UAAU,GAAG,aAAa;UACtB;YACE,UAAU,EAAE,qCAAqC;YACjD,UAAU,EAAE,OAAO;YACnB,SAAS,EAAE;gBACT,KAAK,EAAE,aAAa;aACrB;YACD,YAAY,EAAE;gBACZ,OAAO,EAAE,MAAM;aAChB;YACD,4CAA4C,EAAE;gBAC5C,OAAO,EAAE,MAAM;aAChB;YACD,8CAA8C,EAAE;gBAC9C,OAAO,EAAE,MAAM;aAChB;SACF;UACD,EAAE,CAAA;IAER,MAAM,YACJ,KAAK,OAAA;QACL,eAAe,iBAAA,EACf,QAAQ,EAAE,GAAG,GAAG,OAAO,GAAG,UAAU,EACpC,MAAM,EAAE,KAAK,CAAC,UAAU,GAAG,GAAG,EAC9B,OAAO,EAAE,MAAM,EACf,aAAa,EAAE,QAAQ,EACvB,UAAU,EAAE,YAAY,EACxB,MAAM,EAAE,OAAO,EACf,QAAQ,EAAE,QAAQ,EAClB,SAAS,EAAE,8BAA8B,EACzC,KAAK,EAAE,qCAAiB,CAAC,eAAe,CAAC,CAAC,CAAC,OAAO,EAAE,OAAO,CAAC,CAAC,IAC1D,UAAU,IACb,WAAW,EAAE;YACX,OAAO,EAAE,CAAC;YACV,eAAe,EAAE,0BAA0B;SAC5C,IACF;AACH,CAAC,CAAA;AAED,IAAM,cAAc,GAAqB,UAAC,EAA8B;QAA5B,wBAAS,EAAE,sBAAQ;IAAc,OAAA,6BAAK,SAAS,EAAE,SAAS,IAAG,QAAQ,CAAO;AAA3C,CAA2C,CAAA;AAExH,cAAc,CAAC,YAAY,GAAG;IAC5B,aAAa,EAAE,KAAK;IACpB,aAAa,EAAE,GAAG;IAClB,KAAK,EAAE,EAAE;IACT,GAAG,EAAE,KAAK;CACX,CAAA;AAED,kBAAe,mBAAS,CAAC,cAAc,CAAC,CAAC,KAAK,CAAC,CAAA"}