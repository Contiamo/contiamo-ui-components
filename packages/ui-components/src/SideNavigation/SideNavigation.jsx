// @flow
import React from "react"
import type { Node } from "react"
import glamorous from "glamorous"

import SideNavigationHeader from "./Header/SideNavigationHeader"
import SideNavigationItem from "./Item/SideNavigationItem"
import SideNavigationLink from "./Link/SideNavigationLink"

import { hexOrColor, readableTextColor } from "contiamo-ui-utils"

type Props = {
  className: string,
  children: Node,
  color?: string,
  expandOnHover: boolean,
  expandedWidth: number,
  width: number,
  fix: boolean,
  theme: THEME,
}

const SideNavigation = ({ className, children }: Props): React$Element<*> =>
    <div className={className}>
      {children}
    </div>,
  style = ({
    theme,
    color,
    fix,
    expandOnHover,

    // for some reason, glamorous doesn't get the defaultProps...
    expandedWidth = 280,
    width = 64
  }: Props): {} => {
    const backgroundColor = color
        ? hexOrColor(color)(theme.colors ? theme.colors[color] : "white")
        : theme.colors && theme.colors.primary,
      hoverWidth = expandOnHover
        ? {
          transition: ".3s width ease",
          willChange: "width",
          "&:hover": {
            width: expandedWidth
          },
          "&:hover .Tooltip": {
            display: "none"
          }
        }
        : {}

    return {
      position: fix ? "fixed" : "relative",
      zIndex: (theme.baseZIndex || 0) + 100,
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start",
      width,
      height: "100vh",
      overflow: "hidden",
      padding: theme.spacing * 1.3 || 0,
      backgroundColor,
      color: readableTextColor(backgroundColor)(["black", "white"]),
      ...hoverWidth
    }
  }

SideNavigation.defaultProps = {
  expandOnHover: false,
  expandedWidth: 280,
  width: 64,
  fix: false
}

export default glamorous(SideNavigation)(style)
export { SideNavigationHeader, SideNavigationItem, SideNavigationLink }
