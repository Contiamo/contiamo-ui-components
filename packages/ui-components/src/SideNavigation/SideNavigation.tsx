import * as React from "react"

import glamorous from "glamorous"

import SideNavigationHeader from "./Header/SideNavigationHeader"
import SideNavigationItem from "./Item/SideNavigationItem"
import SideNavigationLink from "./Link/SideNavigationLink"

import { hexOrColor, readableTextColor } from "contiamo-ui-utils"

interface IProps {
  css?: {}
  className?: string
  children?: React.ReactNode
  color?: string
  expandOnHover?: boolean
  expandedWidth?: number
  width?: number
  fix?: boolean
}

const Container = glamorous.div(
  ({
    theme,
    color,
    fix,
    expandOnHover,
    expandedWidth,
    width
  }: {
    theme: Theme
    color?: string
    fix?: boolean
    expandOnHover?: boolean
    expandedWidth: number
    width: number
  }): any => {
    const backgroundColor = color ? hexOrColor(color)(theme.colors.palette[color]) : theme.colors.palette.grey80
    const hoverWidth = expandOnHover
      ? {
          transition: ".3s width cubic-bezier(.8, 0, 0, 1)",
          willChange: "width",
          "&:hover": {
            width: expandedWidth
          }
        }
      : {}

    return {
      width,
      backgroundColor,
      position: fix ? "fixed" : "relative",
      zIndex: theme.baseZIndex + 100,
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start",
      height: "100vh",
      overflow: "hidden",
      boxShadow: "1px 0 2px rgba(0, 0, 0, 0.2)",
      color: readableTextColor(backgroundColor)(["black", "white"]),
      ...hoverWidth,
      "& a:focus": {
        outline: 0,
        backgroundColor: "rgba(255, 255, 255, 0.1)"
      }
    }
  }
)

const SideNavigation: React.SFC<IProps> = ({
  css,
  className,
  children,
  color,
  fix,
  expandOnHover,
  expandedWidth,
  width
}: IProps) => (
  <Container
    css={css}
    className={className}
    color={color}
    fix={fix}
    expandOnHover={expandOnHover}
    expandedWidth={expandedWidth || 240}
    width={width || 60}
  >
    {children}
  </Container>
)

export default SideNavigation
export { SideNavigationHeader, SideNavigationItem, SideNavigationLink }
