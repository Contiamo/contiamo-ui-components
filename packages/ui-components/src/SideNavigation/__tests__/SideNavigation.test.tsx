
import * as React from "react"
import { shallow } from "enzyme"

import SideNavigation from "../SideNavigation"

test("SideNavigation component renders", () => {
  expect(shallow(<SideNavigation />)).toMatchSnapshot()
})
