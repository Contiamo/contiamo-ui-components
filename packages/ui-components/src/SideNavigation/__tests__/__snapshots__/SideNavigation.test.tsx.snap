// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavigation component renders 1`] = `
<SideNavigation
  className="css-nughly"
  expandOnHover={false}
  expandedWidth={280}
  fix={false}
  width={64}
/>
`;
